### PR TITLE
TimeEntries: activity, created_at and spent_on filter/sortby

### PR DESCRIPTION
**Pull request by oliverguenther**
Implements the following filters and sortBy for TimeEntries:

- created_at
- spent_on
- activity(_id)

### CODE REVIEW

File: app/models/queries/time_entries/filters/activity_filter.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    end
+  end
+
+  def type
+    :list_optional
+  end
+
+  def self.key
+    :activity_id
+  end
```
Comment (ulferts) [inline app/models/queries/time_entries/filters/activity_filter.rb:49]: The time entry activity use some really spaced out way of implementing a join table. If a time entry activity is activated in a project, a child of the global activity is created. So time entries that are created receive the specific project time entry activity. The only exception to this is when time entries are configured to be global (active for all projects). 

This data structure has no benefit that I see compared to a simple join table but that is the way the data is currently modelled. 

By the way, for that reason the `TimeEntries::CreateService` has a [mapping code](https://github.com/opf/openproject/blob/dev/app/services/time_entries/create_service.rb#L67) which I noted the `TimeEntries::UpdateService` does not but should have.

File: docs/api/apiv3/endpoints/time_entries.apib [hunk 2]
```diff
@@ -209,11 +209,21 @@ Lists time entries. The time entries returned depend on the filters provided and
 
     + pageSize (optional, integer, `25`) ... Number of elements to display per page.
 
+    + sortBy = ["spent_on", "asc"] (optional, string, `[["speont_on", "asc"]]`) ... JSON specifying sort criteria.
```
Comment (ulferts) [inline docs/api/apiv3/endpoints/time_entries.apib]: typo: "speont_on"

File: docs/api/apiv3/endpoints/time_entries.apib [hunk 3]
```diff
@@ -209,11 +209,21 @@ Lists time entries. The time entries returned depend on the filters provided and
...
+    Accepts the same format as returned by the [queries](#queries) endpoint. Currently supported sorts are:
+      + id: Sort by primary key
+      + hours: Sort by logged hours
+      + spent_on: Sort by spent on date
+      + created_on: Sort by time entry creation datetime
+
```
Comment (ulferts) [inline docs/api/apiv3/endpoints/time_entries.apib:218]: :+1: for adding this section

**Review by ulferts — CHANGES_REQUESTED** [review]
That activity filter needs some more elaborate behaviour because of that weird data structure.

**Comment by oliverguenther** [comment]
@ulferts feel free to implement the changes yourself since I won't be working on it until at least thursday.

**Review by ulferts — APPROVED** [review]

**Comment by oliverguenther** [comment]
Thanks 👏

**Pull request merged**